core/include/traccc/fitting/kalman_filter/gain_matrix_updater.hpp

```diff
@@ -47,9 +47,9 @@ struct gain_matrix_updater {
 
         // Some identity matrices
         // @Note: Make constexpr work
-        static const matrix_type<6, 6> I66 =
+        const matrix_type<6, 6> I66 =
```

Comment:
...
```
/home/beomki/projects/traccc/traccc/core/include/traccc/fitting/kalman_filter/gain_matrix_updater.hpp(52): error: dynamic initialization is not supported for a function-scope static __device__ variable within a __device__/__global__ function
```

Reply:
I tested `constexpr` in algebra-plugins. 
cmath can call `constexpr` functions with small modification, whereas it seems not possible with Eigen. Could you checkout the error log?:
```
/home/beomki/projects/algebra-plugins/algebra-plugins/tests/common/test_host_basics.hpp:232:1:   required from here
/home/beomki/projects/algebra-plugins/algebra-plugins/tests/common/test_host_basics.hpp:314:55: error: the type ‘test_types<float, algebra::eigen::array<float, 2>, algebra::eigen::array<float, 3>, algebra::eigen::array<float, 2>, algebra::eigen::array<float, 3>, algebra::cmath::transform3<algebra::cmath::matrix::actor<int, algebra::eigen::array, algebra::eigen::matrix_type, float, algebra::cmath::matrix::determinant::actor<int, algebra::eigen::matrix_type, float, algebra::cmath::matrix::determinant::partial_pivot_lud<int, algebra::eigen::matrix_type, float, algebra::eigen::math::element_getter>, algebra::cmath::matrix::determinant::hard_coded<int, algebra::eigen::matrix_type, float, algebra::eigen::math::element_getter, 2, 4> >, algebra::cmath::matrix::inverse::actor<int, algebra::eigen::matrix_type, float, algebra::cmath::matrix::inverse::partial_pivot_lud<int, algebra::eigen::matrix_type, float, algebra::eigen::math::element_getter>, algebra::cmath::matrix::inverse::hard_coded<int, algebra::eigen::matrix_type, float, algebra::eigen::math::element_getter, 2, 4> >, algebra::eigen::math::element_getter, algebra::eigen::math::block_getter> >, int, algebra::eigen::matrix_type, algebra::cmath::matrix::actor<int, algebra::eigen::array, algebra::eigen::matrix_type, float, algebra::cmath::matrix::determinant::actor<int, algebra::eigen::matrix_type, float, algebra::cmath::matrix::determinant::partial_pivot_lud<int, algebra::eigen::matrix_type, float, algebra::eigen::math::element_getter>, algebra::cmath::matrix::determinant::hard_coded<int, algebra::eigen::matrix_type, float, algebra::eigen::math::element_getter, 2, 4> >, algebra::cmath::matrix::inverse::actor<int, algebra::eigen::matrix_type, float, algebra::cmath::matrix::inverse::partial_pivot_lud<int, algebra::eigen::matrix_type, float, algebra::eigen::math::element_getter>, algebra::cmath::matrix::inverse::hard_coded<int, algebra::eigen::matrix_type, float, algebra::eigen::math::element_getter, 2, 4> >, algebra::eigen::math::element_getter, algebra::eigen::math::block_getter> >::matrix<3, 2>’ {aka ‘const Eigen::Matrix<float, 3, 2, 0, 3, 2>’} of ‘constexpr’ variable ‘I32’ is not literal
  314 |   constexpr typename TypeParam::template matrix<3, 2> I32 =
      |                                                       ^~~
In file included from /home/beomki/packages/eigen-3.4.0/Eigen/Core:295,
                 from /home/beomki/projects/algebra-plugins/algebra-plugins/math/eigen/include/algebra/math/impl/eigen_getter.hpp:18,
                 from /home/beomki/projects/algebra-plugins/algebra-plugins/math/eigen/include/algebra/math/eigen.hpp:11,
                 from /home/beomki/projects/algebra-plugins/algebra-plugins/frontend/eigen_cmath/include/algebra/eigen_cmath.hpp:10,
                 from /home/beomki/projects/algebra-plugins/algebra-plugins/tests/eigen/eigen_cmath.cpp:9:
/home/beomki/packages/eigen-3.4.0/Eigen/src/Core/Matrix.h:178:7: note: ‘Eigen::Matrix<float, 3, 2, 0, 3, 2>’ is not literal because:
  178 | class Matrix
      |       ^~~~~~
/home/beomki/packages/eigen-3.4.0/Eigen/src/Core/Matrix.h:178:7: note:   ‘Eigen::Matrix<float, 3, 2, 0, 3, 2>’ is not an aggregate, does not have a trivial default constructor, and has no ‘constexpr’ constructor that is not a copy or move constructor
```

Reply:
Let's just go ahead with `const` for now. We'll have a look at why Eigen doesn't like `constexpr` matrices in a next step...